src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,171 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int defaultCapacity = 16;
+    private final float loadFactor = 0.75f;
+    private Node<K, V>[] tables;
+    private int threshold;
+    private int size = 0;
+
+    public MyHashMap() {
+        tables = new Node[defaultCapacity];
+        threshold = (int) (loadFactor * tables.length);
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> newNode = null;
+        int index = index(key);
+        Objects.checkIndex(index, tables.length);
+        if (key == null) {
+            key = (K) Integer.valueOf(index);
+        }
 
+        if (tables[index] == null) {
+            newNode = new Node<>(index, key, value, null);
+            tables[index(key)] = newNode;
+            if (size++ > threshold) {
+                resize();
+            }
+            return;
+        }
+        newNode = new Node<>(index, key, value, null);
+        if (newNode.getHash() == tables[index].hash) {
```

Comment:
```suggestion
        if (getHash(newNode) == tables[index].hash) {
```
lets create getHash(Node) method

Reply:
I use washing methods according to previous recommendations